### PR TITLE
New template Alcatel AOS: `show mac-address-table`

### PR DESCRIPTION
**Pull request by evilmonkey19**

### CODE REVIEW

**Comment by evilmonkey19** [comment]
Added support for R8 version where the command is `show mac-learning`. In the R6 version the command is `show mac-address-table`. I wanted to combine both commands to simplify everything as Alcatel decided to make it difficult :)

File: ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Value PROTOCOL (\S+)
+Value OPERATION (\S+)
+Value INTERFACE (\S+)
+
+Start
+  ^\s*Vlan\s*Mac\s*Address\s*Type\s*Protocol\s*Operation\s*Interface\s*$$
+  ^\s*------+-------------------+--------------+-----------+------------+-----------\s*$$
+  ^\s*${VLAN_ID}\s*${MAC}\s*${TYPE}\s*(---|${PROTOCOL})\s*${OPERATION}\s*${INTERFACE}\s*$$ -> Record
+  ^\s*Domain\s*Vlan\/SrvcId\[ISId\/vnId\]\s*Mac\s*Address\s*Type\s*Operation\s*Interface\s*$$
+  ^\s*------------+----------------------+-------------------+------------------+-------------+-------------------------\s*$$
```
Comment (jmcgill298) [inline ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm:14]:
```suggestion
  ^\s*-+\+
  ^\s*${VLAN_ID}\s*${MAC}\s*${TYPE}\s*(---|${PROTOCOL})\s*${OPERATION}\s*${INTERFACE}\s*$$ -> Record
  ^\s*Domain\s*Vlan\/SrvcId\[ISId\/vnId\]\s*Mac\s*Address\s*Type\s*Operation\s*Interface\s*$$
```

I feel comfortable that we are only matching table dividers with looking for lines that have `-` repeated followed by a `+`

**Review by jmcgill298 — CHANGES_REQUESTED** [review]
just one adjustment @evilmonkey19

**Comment by mjbear** [comment]
> @evilmonkey19 I'd be glad to submit further suggestions if you're willing to accept them via GitHub workflow (Commit Suggestion). Thank you! 🙂

@evilmonkey19 
I'm having issues trying to compare a branch on my fork to a branch on your fork.
Would you please sync your fork's `master` branch as an attempt to fix this?

**Comment by evilmonkey19** [comment]
I synced the fork :)

**Comment by mjbear** [comment]
> I synced the fork now :)

Thank you.
Hmm, GitHub still doesn't want to show me your fork in order for me to submit a PR against your fork.

Oh well, I can work off of the code review page as we've been doing.
Thank you for collaborating with me! :grin:

File: ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Value INTERFACE (\S+)
+
+Start
+  ^\s*Vlan\s*Mac\s*Address\s*Type\s*Protocol\s*Operation\s*Interface\s*$$
+  ^\s*-+\+
+  ^\s*${ADDRESS_NOT_VALID}${VLAN_ID}\s*${MAC}\s*${TYPE}\s*(---|${PROTOCOL})\s*${OPERATION}\s*${INTERFACE}\s*$$ -> Record
+  ^\s*Domain\s*Vlan\/SrvcId\[ISId\/vnId\]\s*Mac\s*Address\s*Type\s*Operation\s*Interface\s*$$
+  ^\s*${DOMAIN}\s*${VLAN_ID}\s*${MAC}\s*${TYPE}\s*${OPERATION}\s*${INTERFACE}\s*$$ -> Record
+  ^\s*Total\s*number\s*of\s*Valid\s*MAC\s*addresses\s*above\s*=\s*\d+\s*$$ -> Record
+  ^\s*
```
Comment (jmcgill298) [inline ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm:17]:
```suggestion
  ^\s*$$
```

File: ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
```
Comment (jmcgill298) [inline ntc_templates/templates/alcatel_aos_show_mac-address-table.textfsm:17]:
```suggestion
  ^Legend
  ^\s+Mac\s+Address:
```
we don't want to catch every line that starts with a space, as that would catch about every line, and negate the point of the catchall/error line.

**Review by jmcgill298 — APPROVED** [review]

**Pull request merged**